model/execution.go

```diff
@@ -41,3 +41,53 @@ type TestFile struct {
 	Content     []byte
 	ChallengeID uint
 }
+
+type TestFileArray []TestFile
+
+func (files TestFileArray) InputFiles() []InputFile {
+	inputs := make([]InputFile, len(files))
+	for index, file := range files {
+		inputs[index] = InputFile{
+			TestFile: file,
+		}
+	}
+	return inputs
+}
+
+func (files TestFileArray) OutputFiles() []OutputFile {
+	outputs := make([]OutputFile, len(files))
+	for index, file := range files {
+		outputs[index] = OutputFile{
+			TestFile: file,
+		}
+	}
+	return outputs
+}
+
+type InputFile struct {
+	TestFile
+}
+
+type InputsArray []InputFile
+
+func (files InputsArray) TestFiles() []TestFile {
+	testFiles := make([]TestFile, len(files))
+	for index, file := range files {
+		testFiles[index] = file.TestFile
+	}
+	return testFiles
+}
```

Comment:
Why are you copying the array without deep copying the test file? Why do you need to copy the array?

Reply:
We can't use the same struct for both inputs and outputs files because ORM doesn't like that and will aggregate inputs and outputs on both fields, so we created a new struct for each. But we also need to work with both inputs and outputs in very similar situations, so we created these two functions to convert from InputsArray/OutputsArray to []TestFile. So, this is just a type conversion from InputsArray to []TestFile, we couldn't find a better way to do this.